WordPress/Classes/Services/SignupService.swift

```diff
@@ -54,8 +54,36 @@ public class SignupService : LocalCoreDataService
                                             success: SignupSuccessBlock,
                                             failure: SignupFailureBlock) {
 
+        // Organize parameters into a struct for easy sharing
         let params = SignupParams(url: url, title: blogTitle, email: emailAddress, username: username, password: password)
-        validateWPComBlogWithParams(params, status: status, success: success, failure: failure)
+
+        // Create call back blocks for the various methods we'll call to create the user account and blog. 
+        // Each success block calls the next step in the process. 
+        // NOTE: The steps below are constructed in reverse order.
```

Comment:
That was a bit weird at first, but I think it's better than nesting them. Sometimes I dream about the `Future` though 😉